### ISSUE TITLE
Unnecessary protobufPackage for type only mode

### ISSUE THREAD

**Issue by mmrath** [stephenh/ts-proto#167]
Thanks for work done here. I have tried different packages for generating types from proto files but this is one that I found best suitable. 

When I am generating only types the generated files export const `protobufPackage`. I am not sure what is the use of this. for me the problem is I wanted to have an index.ts file which re-exports all interfaces from the generated files in one folder. however because of the exported const I am not able use `*` exports. I could do a manual selective export but that would be tedious. Would you have some suggestion or workaround?

`export const protobufPackage = 'mypackage'`

**Comment by VeeeneX** [comment]
I just opened repo to see whether somebody had same problem :smile: 
There's temporary solution, to use named exports/imports:

```ts
export * as Contract from './contract'
export * as Client from './client'
```

Looking at this line I'm afraid there's no option to turn it off.

https://github.com/stephenh/ts-proto/blob/7805699968a7cf201843b84c63f4cdf45d36acd2/src/main.ts#L110

**Comment by stephenh** [comment]
Sure, I'd be cool with a `disableProtobufPackage` type option flag. We're getting a ton of flags ... but ... :shrug:

**Comment by amaurymartiny** [comment]
Related to this issue: most generated files export `DeepPartial`, which creates conflicts on `export *`. Maybe `DeepPartial` could be imported from https://github.com/sindresorhus/type-fest/blob/master/source/partial-deep.d.ts?

**Comment by stephenh** [comment]
I generally prefer the codegen output to have zero dependencies, but an option like `useTypeFestDeepPartial` would be fine, for users to opt-in to that dependency + the shared definition.

**Comment by iangregsondev** [comment]
Yep, agreed, also ran into this too with regards to protobufPackage, being exported from multiple types so you can't do an export * from a barrel :-(

I am trying to think if this is used anywhere with people that are using nestjs ? Anybody?

I can't find a good use case for it on nestjs although I might be missing something.

If this is the case, rather than adding another parameter, we could just not output it when the option is set to nestjs.

Ideas?

**Comment by stephenh** [comment]
> If this is the case, rather than adding another parameter, we could just not output it when the option is set to nestjs.

Yeah, not a bad idea per se.

Maybe another parameter i.e. `disableProtobufPackage` than defaults to true when nestjs is on.

@amaurymartiny had a good point re `DeepPartial` though, i.e. even if we fix not exporting `protobufPackage`, the issue after that will be to conditionally not export `DeepPartial`.

IIRC `DeepPartial` was originally _not_ exported, but then I got a request to explicitly `export` it ... hence being yet another "some users want behavior x / some users want behavior y" sort of thing.

**Comment by stephenh** [comment]
I've added a `exportCommonExports` flag that you can set to `false` (defaults to `true` for backwards compatibility) that will a) not export `DeepPartial` and b) not create the `protobufPackage` `const`.

This allows a minimal reproduction of `import * from ./bar` and `import * from ./foo` to work.

Released in v1.49.0.

**Comment by can-pekesen** [comment]
> I've added a `exportCommonExports` flag that you can set to `false` (defaults to `true` for backwards compatibility) that will a) not export `DeepPartial` and b) not create the `protobufPackage` `const`.
> 
> This allows a minimal reproduction of `import * from ./bar` and `import * from ./foo` to work.
> 
> Released in v1.49.0.

Small hint for all the ones wondering why `exportCommonExports` does not work. It ended up beeing implented as `exportCommonSymbols`.

**Comment by aryankarim** [comment]
@stephenh  `exportCommonSymbols=false` will also remove the exported constant package name as well which ain't common among each proto files. I think they should have kept that.

https://github.com/stephenh/ts-proto/blob/f05de60e4c38c0522669a2ed4202f4cf1267296b/src/main.ts#L179-L193
